src/main/java/com/parsing/pdf/parsing/ParsingPDFService.java

```diff
@@ -0,0 +1,142 @@
+package com.parsing.pdf.parsing;
+
+import com.parsing.exception.PDFParsingException;
+import com.parsing.pdf.parsing.model.Column;
+import com.parsing.pdf.parsing.model.Row;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.sourceforge.tess4j.ITesseract;
+import net.sourceforge.tess4j.Tesseract;
+import net.sourceforge.tess4j.TesseractException;
+import nu.pattern.OpenCV;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.rendering.ImageType;
+import org.apache.pdfbox.rendering.PDFRenderer;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.json.JSONObject;
+import org.opencv.core.Mat;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ParsingPDFService {
+
+    private static final double OFFSET = 5;
+    private static final int WHITE_PIXEL_CODE = 255;
+    private final RectangleDetector rectangleDetector;
+    private final DataRecognizer dataRecognizer;
+    private final TableDetector tableDetector;
+    private final String DIR_TO_READ_TESSDATA = "/tessdata/";
+    private List<Row> table = new ArrayList<>();
+
+    public String parseProzorroFile(MultipartFile file) throws IOException, TesseractException {
+        OpenCV.loadLocally();
+
+        PDDocument document = PDDocument.load(file.getBytes());
+        String lastPagePDF = getLastPagePDF(document);
+        String fileTableName = tableDetector.detectTable(lastPagePDF);
+        table = rectangleDetector.detectRectangles(fileTableName);
+        table = extractTextFromScannedDocument(fileTableName);
+        dataRecognizer.recognizeLotPDFResult(table);
+        JSONObject obj = new JSONObject();
+        obj.put("fileName", file.getOriginalFilename());
+        StringBuilder builder = new StringBuilder();
+        for (Row row : table) {
+            for (Column column : row.getColumns()) {
+                builder.append(column.getParsingResult());
+            }
+        }
+        obj.put("text", builder);
+        return obj.toString();
+    }
+
+    private List<Row> findStrippedTextFromFile(MultipartFile file) {
```

Comment:
should it be something about tables and rows in the method name?

Reply:
this method deleted.